### DISCUSSION TITLE
Is it possible to rebuild a collection using the json files?

### DISCUSSION THREAD

**Discussion by SyahmiRafsan** [nftchef/art-engine#107] [Help + Q&A] [answered]
Hi, I would like to ask if it is possible using this art engine? Thanks a lot!

**Comment by nftchef** [comment, marked as answer]
Hi yes, there is a util script for that, `utils/rebuildAll.js`
My apologies, it looks like it's missing from the documentation.
```
/**
 * Utility for regenerating the same output using the DNA file to
 * redraw each previously generated image.
 *
 * Optionally, you can reconfigure backgrounds,
 * turn off layers, e.g. backgrounds for transparent variations
 * using --omit

 */

Options:
  -o, --omit <omitIndex>         omit any given layer by layer index
  -i, --startIndex <startIndex>  Then num.to start Output naming at, default is 0
  -b, --background <background>  override the config generate background bool
  -s, --source <source>          Optional source path of _dna.json
  -d, --debug                    display additional logging
  -v, --verbose                  display even more additional logging
  -h, --help                     display help for command
```
It requires all layers folders and files to be the same name and in the same locations, but if you modified one png for example (with the same file name) it will rebuild with updated art, which is one of the use cases.

To use it, use the following command with any of the options above
```
node utils/rebuildAll.js
```

**Reply by SyahmiRafsan** [reply]
Thanks for this good ser. God bless you!

**Reply by buhduhzah** [reply]
hey! i was able to remove a trait from my generated collection using removetrait utils. after this, i used `node utils/rebuildAll.js` and its building a (seemingly) different generation. any ideas why this is the case? it also starts building at 0, while my settings in the config.js is for it to start at 1. thanks
